mach_eval/analyzers/electromagnetic/flux_linkage_analyzer.py

```diff
@@ -0,0 +1,141 @@
+import os
+import pandas as pd
+import numpy as np
+from time import time as clock_time
+
+class FluxLinkageJMAG_Problem:
+    """Problem class for processing FEA flux linkages
+    Attributes:
+        toolJmag: JMAG tool required for FEA scripting
+        phase_names: names of phases in machine
+        rated_current: rated current of machine
+    """
+
+    def __init__(self, toolJmag, phase_names, rated_current):
+        self.toolJmag = toolJmag
+        self.phase_names = phase_names
+        self.rated_current = rated_current
+    
+    def add_em_study(self, study_name):
+
+        self.app = self.toolJmag.jd
+        self.model = self.app.GetCurrentModel()
+
+        # Pre-processing
+        self.model.SetName("Machine_FluxLinkage_Project")
+
+        # Initialize JMAG application
+        self.app.SetCurrentStudy(study_name)
+        study = self.model.GetStudy(study_name)
+
+        # Set csv folder output
+        self.results_filepath = os.path.dirname(__file__) + "/run_data/"
+
+        # Create output folder
+        if not os.path.isdir(self.results_filepath):
+            os.makedirs(self.results_filepath)
+
+        # Create csv output file loction
+        study.GetStudyProperties().SetValue("CsvOutputPath", self.results_filepath)
+        study.GetStudyProperties().SetValue("CsvResultTypes", "FEMCoilFlux")
+
+        self.app.SetCurrentStudy(study_name)
+
+        return study
+
+    
+    def del_op_pts(self, study):
+        
+        # Creating circuit function and zeroing all values
+        cs_name = []
+        for i in range(0, len(self.phase_names)):
+            cs_name.append("cs_" + self.phase_names[i]) 
+            f1 = self.app.FunctionFactory().Sin(0, 0.000001, 90)
+            func = self.app.FunctionFactory().Composite()
+            func.AddFunction(f1)
+            study.GetCircuit().GetComponent(cs_name[i]).SetFunction(func)
+            self.app.GetModel(0).GetStudy(0).GetDesignTable().AddParameterVariableName("cs_%s (CurrentSource): 1 (Composite Function): Amplitude" % self.phase_names[i])
+            self.app.GetModel(0).GetStudy(0).GetDesignTable().AddCase()
+            self.app.GetModel(0).GetStudy(0).GetDesignTable().SetValue(0, i, 0)
+
+    def new_op_pt(self, I_hat, i):
```

Comment:
Let's rename to `new_operating_point(self)` and have it return an index for the new operating point (like my pseudocode does [here](https://github.com/Severson-Group/eMach/issues/296#issuecomment-2142904755))